notes.go

```diff
@@ -0,0 +1,207 @@
+	"context"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+type notesHandler struct {
+	notesClient notesv1.NotesAPIClient
+}
+
+func (h *notesHandler) CreateNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.CreateNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		return
+	}
+
+	res, err := h.notesClient.CreateNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) GetNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.GetNoteRequest{
+		Id: c.Param("note_id"),
+	}
+
+	res, err := h.notesClient.GetNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) UpdateNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.UpdateNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		c.JSON(http.StatusOK, httpError{Error: err.Error()})
+		return
+	}
+	body.Id = c.Param("note_id")
+
+	res, err := h.notesClient.UpdateNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) DeleteNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.DeleteNoteRequest{
+		Id: c.Param("note_id"),
+	}
+
+	res, err := h.notesClient.DeleteNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) ListNotes(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.ListNotesRequest{
+		AuthorId: c.Param("author_id"),
+	}
+
+	res, err := h.notesClient.ListNotes(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) ExportNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.ExportNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		return
+	}
+	body.NoteId = c.Param("note_id")
```

Comment:
There should be `body.ExportFormat` aswell.
I guess we could put it in query param as it is a GET request.
`c.Request.URL.Query()` returns a map[string]string of query params.

Reply:
I need a uint32 for export type not a string thats why I didn't put the format in query param. If you have a solution excepted : 
```go
formatStr := c.Query("format")
	var format uint32
	if formatStr == notesv1.NoteExportFormat_NOTE_EXPORT_FORMAT_INVALID.String() {
		err = errors.New("Wrong format for exporting note")
		writeError(c, http.StatusBadRequest, err)
		return
	}
	if formatStr == notesv1.NoteExportFormat_NOTE_EXPORT_FORMAT_MARKDOWN.String() {
		format = 1
	}
	if formatStr == notesv1.NoteExportFormat_NOTE_EXPORT_FORMAT_PDF.String() {
		format = 2
	}```
	You're welcome

Reply:
`strconv.Atoi(c.Query("format"))` can convert a string into int and tell you with `err` if it failed or not